pandas/core/frame.py

```diff
@@ -2033,7 +2033,7 @@ def _ixs(self, i, axis=0):
                 return self.loc[:, lab_slice]
             else:
                 if isinstance(label, Index):
-                    return self.take(i, axis=1, convert=True)
+                    return self.take(i, axis=1)
```

Comment:
*every* reference to an internal take should be to ``._take`` (this is pretty much all of them)

Reply:
Is this really needed? 
Can't we keep `take` and only use `_take` where it is needed? IMO it just makes it more complex to understand the code.

Reply:
Agree with @jorisvandenbossche .  I'm not sure I really understand what you mean by "reference to an internal take" since the original references are to a public function.

Reply:
you have now 2 functions to do the same thing , a public and private version

it is much more clear to simply use _take internally in all cases

the public function is just an interface 

otherwise a future reader will not know which to use - recipe for confusion

pls change all uses to _take

Reply:
There will be confusion anyhow, because then people will wonder what the difference is with pubkic take, and in majority of the cases, there is *no* difference. IMO that is more clear by using the public function.

Reply:
> in majority of the cases, there is no difference

My sentiment exactly.  There is currently only one case where I had to use `._take`  We use `.take()` internally almost everywhere else.  Unless you want to overhaul *all* of the internal uses (beyond those touched upon in this PR) + duplicate implementations of `.take()` as `._take()` for non-`Series` and `DataFrame` objects, I have a hard time believing that that will work out well.

Reply:
There is AFAIK only one class where negative indexing isn't handled properly, and that is `Categorical`.  Makes we wonder whether we should revisit the option of adding supporting given this disagreement about design though (@chris-b1 thoughts?)

Reply:
Let me boil it down to this: besides potential test failures, is there reason why we call `take_1d` in `pandas.core.categorical` instead of just calling `self._codes.take(...)` ?

Reply:
pls change all references to _take

.take is ONLY a wrapper function and should not be used anywhere in the code base
this eliminates any possibility of confusion and is inline with the existing style
mixing usages is cause for trouble


your other references are simply not relevant here and are again purely internal uses (take_1d)
if u want to bring them up in another issue by all means

let's keep the scope to changing refs to the NDFrame .take changes

Reply:
If there were multiple places where I *had* to use `._take()`, then I think I would be more inclined to agree with you @jreback .  However, at this point, there is only *one* place where this had to happen, and if necessary, I can easily "clear up" any confusion by adding a comment.

IMO one usage isn't convincing enough for me to convert all of other invocations to use `._take()`.

Reply:
@gfyoung if this were a new public function, and you wrote a private impl ``_take`` as well. Then you certainly would use the private impl everywhere. I don't see a difference here. Pls make the change.